pkg/server/test/read_changes.go

```diff
@@ -265,3 +287,32 @@ func setup(store string, datastore storage.OpenFGADatastore) (context.Context, s
 
 	return ctx, datastore, nil
 }
+
+// writeTuplesConcurrently writes two groups of tuples concurrently to expose potential race issues when reading changes
+func writeTuplesConcurrently(t *testing.T, store string, datastore storage.OpenFGADatastore, tupleGroupOne, tupleGroupTwo []*openfgav1.TupleKey) (context.Context, storage.ChangelogBackend) {
+	t.Helper()
+	ctx := context.Background()
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		err := datastore.Write(ctx, store, []*openfgav1.TupleKey{}, tupleGroupOne)
+		if err != nil {
+			t.Logf("failed to write tuples: %s", err)
```

Comment:
Instead of Logf, should it be Errorf?

Reply:
No, `Errorf` calls `FailNow` and 

> FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test.

[source](https://pkg.go.dev/testing#:~:text=c%20*B)%20FailNow()-,FailNow%20marks%20the%20function%20as%20having%20failed%20and%20stops%20its%20execution,during%20the%20test.%20Calling%20FailNow%20does%20not%20stop%20those%20other%20goroutines.,-func%20(*B)

Reply:
If write fails, does it render test invalid and we should fail?